### DISCUSSION TITLE
SPA support

### DISCUSSION THREAD

**Discussion by yugeshananddbg** [freshframework/fresh#2652] [Q&A]
I'm really disappointed to see such a great framework without SPA support. It's better to build your website using Next.js or another modern framework. Fresh websites look like they were built with PHP or outdated technology. It's really frustrating.

**Comment by predaytor** [comment]
Fresh ideology clearly fits with server-first approach relying on small runtime footprint using islands for interactivity, similar to Next.js App Router, e.g. client components, without the fully-featured client-side router, which is indeed required for SPA. 

There are a lot of issues as the framework is still in development, especially regarding support for styling solutions and plugins (compared to Vite), but I'd say it has the cleanest and simplest API. Using JSX as a templating language, aka server components, is the best thing ever, and with @preact/signals, state management between components is very neat. It is really suitable for content oriented websites and not much for app development.

**Reply by predaytor** [reply]
I would say I see so much potential, especially using the Deno ecosystem compared to Node. 

Astro is currently better for almost everything, but it's clearly overcomplicated with a lot of APIs, and I don't like the special syntax they use. Give it more time and I would 100% choose Fresh over Remix/Next.js for most projects that have simpler requirements.

**Reply by yugeshananddbg** [reply]
Is it possible to build my Single Page Application (SPA) with Fresh JS? I've noticed that SPAs tend to feel faster from the user's perspective, even if the backend is slow.

**Reply by predaytor** [reply]
I don't think it's possible. For SPA I'd recommend Remix (without SSR) or React + Vite + tanstack/query.

**Comment by CAYdenberg** [comment]
Then use Next.

**Reply by yugeshananddbg** [reply]
Thanks great suggestion!
Next overall better option to choose.
I have very poor experience with Preact and Fresh. It feels too outdated in term of usability (Without spa).
